boards/components/src/can.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 OxidOS Automotive SRL
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//
+// Author: Teona Severin <teona.severin@oxidos.io>
+
+//! Component for CAN syscall interface.
+//!
+//! This provides one Component, `CanComponent`, which implements a
+//! userspace syscall interface to the Can peripheral.
+//!
+//! Usage
+//! -----
+//! ```rust
+//! let can = components::can::CanComponent::new(
+//!     board_kernel,
+//!     capsules::can::DRIVER_NUM,
+//!     &peripherals.can1
+//! ).finalize(components::can_component_static!(
+//!     stm32f429zi::can::Can<'static>
+//! ));
+//! ```
+//!
+
+use capsules::can::CanCapsule;
+use core::mem::MaybeUninit;
+use kernel::component::Component;
+use kernel::hil::can;
+use kernel::{capabilities, create_capability};
+
+#[macro_export]
+macro_rules! can_component_static {
+    ($C:ty $(,)?) => {{
+        use capsules::can::CanCapsule;
+        use core::mem::MaybeUninit;
+        use kernel::hil::can;
+
+        static mut CAN_TX_BUF: [u8; can::STANDARD_CAN_PACKET_SIZE] =
+            [0; can::STANDARD_CAN_PACKET_SIZE];
+        static mut CAN_RX_BUF: [u8; can::STANDARD_CAN_PACKET_SIZE] =
+            [0; can::STANDARD_CAN_PACKET_SIZE];
```

Comment:
The updated convention is to use `static_buf!()` for all globals, even byte buffers